src/game/entities/Player/Player.ts

```diff
@@ -0,0 +1,208 @@
+
+  //#region
+  public computeNewAcceleration(currentAcc: number, variation: number): number {
+    let ca = currentAcc + variation * ACCELERATION_INTENSITY_MULTIPLAYER;
+    const maxAcceleration = this.domainPlayer.maxAcceleration;
+
+    if (Math.abs(ca) > maxAcceleration) {
+      ca = ca < 0 ? -maxAcceleration : maxAcceleration;
+    }
+
+    return ca;
+  }
+
+  public applyRestitution(delta: number): void {
+    const slowFactor = delta * this.domainPlayer.friction;
+
+    if (this.vx != 0 && this.ax === 0) {
+      this.updateValue(
+        'vx',
+        this.vx + (this.vx < 0 ? slowFactor : -slowFactor)
+      );
+    }
+    if (this.vy != 0 && this.ay === 0) {
+      this.updateValue(
+        'vy',
+        this.vy + (this.vy < 0 ? slowFactor : -slowFactor)
+      );
+    }
+  }
+
+  public applyAcceleration(delta: number): void {
+    let cvx = this.vx + this.ax * delta;
+    let cvy = this.vy + this.ay * delta;
+
+    if (cvx === this.vx && cvy === this.vy) {
+      return;
+    }
+
+    const maxSpeed = this.domainPlayer.maxSpeed;
+
+    if (Math.abs(this.vx) > maxSpeed) {
+      cvx = cvx < 0 ? -maxSpeed : maxSpeed;
+    }
+
+    if (Math.abs(this.vy) > maxSpeed) {
+      cvy = cvy < 0 ? -maxSpeed : maxSpeed;
+    }
+
+    this.updateValue('vx', cvx);
+    this.updateValue('vy', cvy);
+  }
+
+  public checkThresholds(): void {
+    if (Math.abs(this.vx) < PLAYER_MIN_VELOCITY) {
+      this.updateValue('vx', 0);
+    }
+    if (Math.abs(this.vy) < PLAYER_MIN_VELOCITY) {
+      this.updateValue('vy', 0);
+    }
+
+    if (Math.abs(this.ax) < PLAYER_MIN_VELOCITY) {
+      this.updateValue('ax', 0);
+    }
+    if (Math.abs(this.ay) < PLAYER_MIN_VELOCITY) {
+      this.updateValue('ay', 0);
+    }
+  }
+
+  public update(delta: number): PlayerBasicUpdate {
+    this.dirtyList = new Set();
+
+    this.applyRestitution(delta);
+    this.applyAcceleration(delta);
+    this.checkThresholds();
+
+    if (
+      this.vx === 0 &&
+      this.vy === 0 &&
+      Math.abs(this.previousPlayerLookAtRad - this.playerLookAtRad) <
+        PLAYER_MIN_ROTATION
+    ) {
+      return null;
+    }
+
+    this.previousPlayerLookAtRad = this.playerLookAtRad;
+
+    // Move with set velocity
+    this.x += this.vx * delta;
+    this.y += this.vy * delta;
+
+    return [
+      this.id,
+      toFixedIfNecessary(this.x, 2),
+      toFixedIfNecessary(this.y, 2),
+      toFixedIfNecessary(this.playerLookAtRad, 3),
+    ] as const;
+  }
+
+  public static createRandom(world: World, x?: number, y?: number): Player {
+    const cx = x ? x : getRandomInt(50, world.width - 50);
+    const cy = y ? y : getRandomInt(50, world.height - 50);
+    const sizeMultiplier = getRandomInt(10, 100);
+    const speed = getRandomInt(30, 60);
+
+    return new Player(
+      null,
+      cx,
+      cy,
+      sizeMultiplier / 2,
+      sizeMultiplier,
+      getRandomInt(-speed, speed),
+      getRandomInt(-speed, speed),
+      20
+    );
+  }
+
+  static indexToKey = [
```

Comment:
```suggestion
  static indexToKey = Object.keys(keyToIndex);
```

Since it is static it will only be calculated once